src/transformers/quantizers/quantizer_awq.py

```diff
@@ -52,8 +52,12 @@ def validate_environment(self, device_map, **kwargs):
         if not is_accelerate_available():
             raise ImportError("Loading an AWQ quantized model requires accelerate (`pip install accelerate`)")
 
-        if self.quantization_config.version == AWQLinearVersion.GEMM and not torch.cuda.is_available():
-            logger.warning_once("No CUDA found, replace GEMM with IPEX version to support non-cuda AWQ model.")
+        if (
+            self.quantization_config.version == AWQLinearVersion.GEMM
+            and not torch.cuda.is_available()
+            and not torch.xpu.is_available()
+        ):
+            logger.warning_once("No CUDA or XPU found, replace GEMM with IPEX version to support non-cuda AWQ model.")
```

Comment:
small suggestion to replace `non-cuda`
```suggestion
        if (
            self.quantization_config.version == AWQLinearVersion.GEMM
            and not torch.cuda.is_available()
            and not torch.xpu.is_available()
        ):
            logger.warning_once("No CUDA or XPU found, consider switching to the IPEX version for CPU-only execution.")
```